### PR TITLE
Change how merged PR commit info are prepared

### LINKED ISSUE #3222: Keep merged PR info in hidden branch
To make merged PR commit tab more stable and consistent for each PR there should be hidden branch created same as github does it - `refs/pull/x/head`

### PR DESCRIPTION
**Pull request by lafriks**
Fixes PR commit and diff display if PR is merged with squash or rebase. Fixes #3222

### CODE REVIEW

**Comment by lafriks** [comment]
Depends on go-gitea/git#107

**Review by ethantkoenig — APPROVED** [review]

File: routers/repo/pull.go [hunk 1]
```diff
@@ -253,40 +253,30 @@ func setMergeTarget(ctx *context.Context, pull *models.PullRequest) {
...
-	if err != nil {
-		ctx.ServerError("Repo.GitRepo.CommitsCountBetween", err)
-		return
-	}
-	ctx.Data["NumFiles"], err = ctx.Repo.GitRepo.FilesCountBetween(pull.MergeBase, endCommitID.String())
 	if err != nil {
-		ctx.ServerError("Repo.GitRepo.FilesCountBetween", err)
-		return
+		if strings.Contains(err.Error(), "fatal: Not a valid object name") {
+			ctx.Data["IsPullReuqestBroken"] = true
```
Comment (thehowl) [inline routers/repo/pull.go:267]: Looks like there's a typo here (`Reuqest`), although it is spread in the entire codebase. Can you change it now or do you think it's better to make another PR after this one is merged?

**Review by thehowl — COMMENTED** [review]
Just a minor note

File: routers/repo/pull.go (hunk 1 above)
Comment (ethantkoenig) [reply routers/repo/pull.go:267]: Nice catch, IMO a separate PR would be better

Comment (lafriks) [reply routers/repo/pull.go:267]: I also think it should be separate PR as this one fixes different issue and that change would make this one harder to review

**Comment by thehowl** [comment]
Alright. It's a LGTM from me then.
